sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java

```diff
@@ -133,17 +140,27 @@ public void populateDisplayData(DisplayData.Builder builder) {
         .include("source", source);
   }
 
-  private static class UnboundedToBoundedSourceAdapter<T>
+  /**
+   * An Adapter wraps the underlying {@link UnboundedSource} with the specified bounds on
+   * number of records and read time into {@link BoundedSource}.
+   */
+  @AutoValue
+  public abstract static class UnboundedToBoundedSourceAdapter<T>
```

Comment:
Making it private will fail compilation. The reason would be Generated AutoValue annotation can't access private class here.